src/condor_utils/env.cpp

```diff
@@ -755,3 +759,43 @@ void WhiteBlackEnvFilter::ClearWhiteBlackList() {
 	m_white.clear();
 }
 
+/*static*/ char* Env::GetProcessEnvBlock(pid_t pid, size_t size_max, int & error)
+{
+	char * envblock = nullptr;
+	std::string filename;
+#ifdef WIN32
+	CSysinfo sysinfo;
+	DWORD dw = S_OK;
+	envblock = sysinfo.GetProcessEnvironment(pid, size_max, dw);
+	error = (int)dw;
+	return envblock;
+#else
+	filename = "/proc/" + std::to_string(pid) + "/environ";
+#endif
+
+	int fd = safe_open_wrapper_follow(filename.c_str(), O_RDONLY | _O_BINARY, 0600);
+	if (fd < 0) {
+		error = errno;
+		dprintf(D_ALWAYS, "Failed to open environment %s for read: %d %s\n", filename.c_str(), error, strerror(errno));
+		return nullptr;
+	}
+
+	const int max_envblock = 64*1024*1024; // 64 Mb absolute max.
+	struct stat statbuf = {};
+	fstat(fd, &statbuf);
```

Comment:
From my testing, stat()ing /proc/pid/environ will always show an empty file. So this call is pointless.

Reply:
useful when I was simulating on windows, but I might as well remove it and change the code to scan for the end...